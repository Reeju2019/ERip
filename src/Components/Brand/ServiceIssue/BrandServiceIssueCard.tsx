import * as React from 'react'
import './ServiceIssue.css'

interface IBrandServiceIssueCardProps {
  data: {
    problem: string
    image: string
  }
}

const BrandServiceIssueCard: React.FunctionComponent<IBrandServiceIssueCardProps> = ({ data }) => {
  return (
    <>
      <div className='d-flex flex-column m-2'>
        <div className='p-2 bg-white '>
          <div className='service-img'>
            <img
              src={data?.image}
              className='images d-flex justify-content-center'
              alt={data?.problem}
            />
          </div>
        </div>
        <p className='service-para'>{data?.problem}</p>
      </div>
    </>
  )
}

export default BrandServiceIssueCard
