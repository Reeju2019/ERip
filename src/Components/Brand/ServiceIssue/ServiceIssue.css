.service-issue {
  background-color: #f5f6fa;
  margin: 0 !important;
}

.fontawesome {
  color: red;
  background-color: green;
}

.slick-prev:before {
  color: rgb(6, 9, 177);
}
.slick-next:before {
  color: blue;
}

/* issue card */
.service-img {
  max-width: 2vw;
}

.images {
  width: 5vmax;
  height: 6vmax;
}

.service-para {
  font-size: 1vmax;
  display: flex;
  justify-content: center;
}
