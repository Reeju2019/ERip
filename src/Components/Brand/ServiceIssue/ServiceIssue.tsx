import React from 'react'
import BrandPageData from '../../../Data/BrandPage.mock.json'
import BrandServiceIssueCard from './BrandServiceIssueCard'
import './ServiceIssue.css'
import Carousel from 'react-multi-carousel'
import 'react-multi-carousel/lib/styles.css'

const ServiceIssue: React.FunctionComponent = () => {
  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 4,
      slidesToSlide: 1,
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
      slidesToSlide: 1,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 2,
      slidesToSlide: 1,
    },
  }
  return (
    <>
      <section className='service-issue'>
        <div className='container'>
          <div className='row'>
            <div className='col-12'>
              <div className='center-title mb-0'>
                <h3>
                  <strong>Service Issues</strong>
                </h3>
              </div>
            </div>
          </div>

          <div className='row d-flex justify-content-center pb-4'>
            <div className='col-8 service-content'>
              <Carousel
                swipeable={true}
                draggable={true}
                showDots={true}
                responsive={responsive}
                ssr={true}
                infinite={true}
                autoPlay={false}
                autoPlaySpeed={1000}
                keyBoardControl={true}
                customTransition='all .5'
                transitionDuration={500}
                containerClass='carousel-container'
                removeArrowOnDeviceType={['tablet', 'mobile']}
                dotListClass='custom-dot-list-style'
                itemClass='carousel-item-padding-40-px'
                partialVisible={true}
                rewindWithAnimation={true}
                sliderClass='react-multi-carousel-track'
              >
                {BrandPageData.brandServiceIssue.map((item, index) => {
                  return (
                    <div key={index} className='carousel-item'>
                      <BrandServiceIssueCard data={item} />
                    </div>
                  )
                })}
              </Carousel>
            </div>
          </div>
        </div>
      </section>
    </>
  )
}

export default ServiceIssue
