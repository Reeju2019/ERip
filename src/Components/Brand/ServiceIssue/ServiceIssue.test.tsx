import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ServiceIssue from './ServiceIssue'

test('service issue', () => {
  render(<ServiceIssue />)
})

describe('App rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<ServiceIssue />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<ServiceIssue />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
