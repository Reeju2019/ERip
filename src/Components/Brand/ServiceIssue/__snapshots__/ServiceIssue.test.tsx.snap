// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering specification SnapShot testing 1`] = `
<section
  className="service-issue"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <div
          className="center-title mb-0"
        >
          <h3>
            <strong>
              Service Issues
            </strong>
          </h3>
        </div>
      </div>
    </div>
    <div
      className="row d-flex justify-content-center pb-4"
    >
      <div
        className="col-8 service-content"
      >
        <div
          className="react-multi-carousel-list carousel-container "
          dir="ltr"
        >
          <ul
            className="react-multi-carousel-track react-multi-carousel-track"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "overflow": "unset",
                "transform": "translate3d(0px,0,0)",
                "transition": "none",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
